Can't build master with go1.10

#### Overview of the Issue

According to Contributing guidelines, we should be able to build with go1.10+ (https://github.com/hashicorp/consul/blob/master/.github/CONTRIBUTING.md#building-consul)

But the current master can't be built with g1.10+

#### Reproduction Steps

Steps to reproduce this issue, eg:

```
➜  consul git:(master) go version
go version go1.10.3 darwin/amd64
➜  consul git:(master) make dev
--> Making [GH-xxxx] references clickable...
==> Building Consul - OSes: darwin, Architectures: amd64
Building sequentially with go install
--->   darwin/amd64
# github.com/hashicorp/consul/agent/connect
agent/connect/parsing.go:205:6: undefined: strings.ReplaceAll
ERROR: Failed to build Consul for darwin/amd64
make: *** [dev-build] Error 1
```
Maybe the contributing guide should be updated?

--- a/.github/CONTRIBUTING.md
+++ b/.github/CONTRIBUTING.md
@@ -60,7 +60,7 @@ issue. Stale issues will be closed.
 ## Building Consul
 
 If you wish to work on Consul itself, you'll first need [Go](https://golang.org)
-installed (version 1.10+ is _required_). Make sure you have Go properly installed,
+installed (version 1.12+ is _required_). Make sure you have Go properly installed,
 including setting up your [GOPATH](https://golang.org/doc/code.html#GOPATH).
 
 Next, clone this repository into `$GOPATH/src/github.com/hashicorp/consul` and 

